core/modules/replica/WorkerServerConnection.cc

```diff
@@ -0,0 +1,543 @@
+                    proto::ReplicationResponseReplicate response;
+                    _processor.checkStatus(hdr.id(),
+                                           request,
+                                           response);
+                    reply(hdr.id(),
+                          response);
+                    break;
+                }
+                case proto::ReplicationReplicaRequestType::REPLICA_DELETE: {
+                    proto::ReplicationResponseDelete response;
+                    _processor.checkStatus(hdr.id(),
+                                           request,
+                                           response);
+                    reply(hdr.id(),
+                          response);
+                    break;
+                }
+                case proto::ReplicationReplicaRequestType::REPLICA_FIND: {
+                    proto::ReplicationResponseFind response;
+                    _processor.checkStatus(hdr.id(),
+                                           request,
+                                           response);
+                    reply(hdr.id(),
+                          response);
+                    break;
+                }
+                case proto::ReplicationReplicaRequestType::REPLICA_FIND_ALL: {
+                    proto::ReplicationResponseFindAll response;
+                    _processor.checkStatus(hdr.id(),
+                                           request,
+                                           response);
+                    reply(hdr.id(),
+                          response);
+                    break;
+                }
+            }
+            break;
+        }
+        default:
+            throw std::logic_error(
+                  "WorkerServerConnection::processManagementRequest() unhandled request type: '" +
+                  proto::ReplicationManagementRequestType_Name(hdr.management_type()));
+    }
+}
+
+void WorkerServerConnection::processServiceRequest(proto::ReplicationRequestHeader& hdr) {
+
+    proto::ReplicationServiceResponse response;
+
+    // All performance counters for this type of requests should be
+    // equal because this is the instantenous request
+
+    WorkerPerformance performance;
+    performance.setUpdateStart();
+    performance.setUpdateFinish();
+    response.set_allocated_performance(performance.info());
+
+    switch (hdr.service_type()) {
+
+        case proto::ReplicationServiceRequestType::SERVICE_SUSPEND: {
+
+            // This operation is allowed to be asynchronious as it may take
+            // extra time for the processor's threads to finish on-going processing
+
+            _processor.stop();
+            _processor.setServiceResponse(
+                  response,
+                  hdr.id(),
+                  _processor.state() == WorkerProcessor::State::STATE_IS_RUNNING ?
+                      proto::ReplicationServiceResponse::FAILED :
+                      proto::ReplicationServiceResponse::SUCCESS);
```

Comment:
this function call is hard to read.

Reply:
It looks good to me. but I made a minor improvement which may help others to read the statement.